src/bin/tools/unzip.ts

```diff
@@ -1,184 +1,89 @@
-import { createReadStream, createWriteStream } from "fs";
-import { mkdir, stat, unlink } from "fs/promises";
-import { dirname as pathDirname, join as pathJoin, relative as pathRelative } from "path";
-import { type Readable } from "stream";
-import { createInflateRaw } from "zlib";
-import { partitionPromiseSettledResults } from "./partitionPromiseSettledResults";
-
-export type MultiError = Error & { cause: Error[] };
-
-/**
- * Extract the archive `zipFile` into the directory `dir`. If `archiveDir` is given,
- * only that directory will be extracted, stripping the given path components.
- *
- * If dir does not exist, it will be created.
- *
- * If any archive file exists, it will be overwritten.
- *
- * Will unzip using all available nodejs worker threads.
- *
- * Will try to clean up extracted files on failure.
- *
- * If unpacking fails, will either throw an regular error, or
- * possibly an `MultiError`, which contains a `cause` field with
- * a number of root cause errors.
- *
- * Warning this method is not optimized for continuous reading of the zip
- * archive, but is a trade-off between simplicity and allowing extraction
- * of a single directory from the archive.
- *
- * @param zipFilePath the file to unzip
- * @param extractDirPath the target directory
- * @param pathOfDirToExtractInArchive if given, unpack only files from this archive directory
- * @throws {MultiError} error
- * @returns Promise for a list of full file paths pointing to actually extracted files
- */
-export async function unzip(zipFilePath: string, extractDirPath: string, pathOfDirToExtractInArchive?: string): Promise<string[]> {
-    const dirsCreated: (string | undefined)[] = [];
-    dirsCreated.push(await mkdir(extractDirPath, { recursive: true }));
-    const promises: Promise<string>[] = [];
-
-    // Iterate over all files in the zip, skip files which are not in archiveDir,
-    // if given.
-    for await (const record of iterateZipArchive(zipFilePath)) {
-        const { path: recordPath, createReadStream: createRecordReadStream } = record;
-        if (pathOfDirToExtractInArchive && !recordPath.startsWith(pathOfDirToExtractInArchive)) {
-            continue;
+import fsp from "node:fs/promises";
+import fs from "fs";
+import path from "node:path";
+import yauzl from "yauzl";
+import stream from "node:stream";
+import { promisify } from "node:util";
+
+const pipeline = promisify(stream.pipeline);
+
+async function pathExists(path: string) {
```

Comment:
this is already present in downloadAndUnzip.ts, maybe better  extract?

Reply:
💯

Reply:
actually I didn´t want to touch anything other than the `unzip.ts`. Adding something like a `fs-utils` folder wouldn´t be a bad idea.